File: src/execgo_playground/adapters/autogen_adapter.py
from __future__ import annotations

from importlib.util import find_spec

from ..models import AdapterCapabilities, PlanContext, StandardPlan
from .base import BaseAdapter


class AutoGenAdapter(BaseAdapter):
    framework = "autogen"

    def capabilities(self) -> AdapterCapabilities:
        native = find_spec("autogen") is not None or find_spec("pyautogen") is not None
        notes = ["Uses a normalized AutoGen-style conversation trace."]
        if not native:
            notes.append("autogen package not installed; using platform fallback planner.")
        return AdapterCapabilities(framework=self.framework, native_available=native, notes=notes)

    def plan(self, context: PlanContext) -> StandardPlan:
        return self._plan_from_provider(
            context,
            {
                "agents": ["planner", "reviewer", "executor"],
                "conversation": [
                    {"speaker": "planner", "intent": "draft_standard_plan"},
                    {"speaker": "reviewer", "intent": "check_execgo_constraints"},
                    {"speaker": "executor", "intent": "emit_standard_plan"},
                ],
            },
        )

    def replay(self, trace: dict, context: PlanContext) -> StandardPlan:
        self._last_trace = trace
        if "plan" in trace:
            return StandardPlan.model_validate(trace["plan"]).model_copy(update={"framework": self.framework, "mode": "replay"})
        return self._load_reference_plan(context).model_copy(update={"framework": self.framework, "mode": "replay"})

File: src/execgo_playground/adapters/autogen_adapter.ts
import { normalizeStandardPlan, type AdapterCapabilities, type PlanContext, type StandardPlan } from "../models.js";
import { BaseAdapter, fallbackCapabilities } from "./base.js";

export class AutoGenAdapter extends BaseAdapter {
  framework = "autogen";

  capabilities(): AdapterCapabilities {
    return fallbackCapabilities(this.framework, [
      "Uses a normalized AutoGen-style conversation trace.",
      "Native package probing removed in the TypeScript control plane; using platform fallback planner.",
    ]);
  }

  plan(context: PlanContext): Promise<StandardPlan> {
    return this.planFromProvider(context, {
      agents: ["planner", "reviewer", "executor"],
      conversation: [
        { speaker: "planner", intent: "draft_standard_plan" },
        { speaker: "reviewer", intent: "check_execgo_constraints" },
        { speaker: "executor", intent: "emit_standard_plan" },
      ],
    });
  }

  replay(trace: Record<string, unknown>, context: PlanContext): StandardPlan {
    this.lastTraceValue = trace;
    if ("plan" in trace) {
      return normalizeStandardPlan(trace.plan, { framework: this.framework, mode: "replay" });
    }
    return normalizeStandardPlan(this.loadReferencePlan(context), { framework: this.framework, mode: "replay" });
  }
}
